lib/composer.js

```diff
@@ -44,10 +48,11 @@ function buildDependencies(
     if (applicationData) {
       depFoundVersion = getVersion(applicationData);
     } else {
-      // we couldn't find the dependency version in the lock
       // here we use the version from the requires - not a locked version
-      depFoundVersion = _.get(_.get(composerJsonObj, 'require'), depName) ||
-        _.get(requires, depName);
+      depFoundVersion =
+        _.get(systemVersions, depName,
+          _.get(_.get(composerJsonObj, 'require'), depName,
+            _.get(requires, depName)));
```

Comment:
Maybe just a combination of `_.get(...) || _.get(...) || _.get(...)`? Will be easier to read, this nesting is a bit tough to get through.